### PR TITLE
Fix for issue #990, implements the PBMTE bit in the henvcfg /menvcfg reisters

### LINKED ISSUE #990: PBMTE bit not implemented in menvcfg and henvcfg
`PBMTE` needs to be added to both registers, `menvcfg` and `henvcfg`, and checked during TableWalks to determine if the `PBMT` bits can be set for leaf PTEs (see c4fdc8fe95e497ab6e107117a7de47ec40e7de7d). Currently Spike only checks to see if the extension is enabled.

In addition, the spec notes that `PBMT=3` is a reserved value, but doesn't specify behavior for that case. Currently Spike does nothing for this case (proceeds as if `PBMT=0`). Should a page fault be thrown in this case, as with other reserved values?

### PR DESCRIPTION
**Pull request by rbuchner-aril**
- Implements the `PBMTE` bit in the `henvcfg` /`menvcfg` registers and checks them during TableWalks
- Makes the value `PBMT==3`, which is reserved in the spec, cause a fault during a TableWalk
- Makes the reset value of `*envcfg.PBMTE = 1` for backward compatibility

### CODE REVIEW

File: riscv/mmu.cc [hunk 1]
```diff
@@ -299,7 +299,7 @@ reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, access_type type, access_type trap_ty
         break;
       } else if (!proc->extension_enabled(EXT_SVNAPOT) && (pte & PTE_N)) {
         break;
-      } else if (!proc->extension_enabled(EXT_SVPBMT) && (pte & PTE_PBMT)) {
+      } else if (!(proc->get_state()->csrmap[CSR_HENVCFG]->read() & HENVCFG_PBMTE) && (pte & PTE_PBMT)) {
```
Comment (scottj97) [inline riscv/mmu.cc]: Prefer accessing the CSR object directly, i.e. `proc->get_state()->henvcfg->read()`. The `csrmap` is there for dynamic access such as `csrw` instruction execution where the specific CSR is unknown at Spike compile time.

**Review by scottj97 — CHANGES_REQUESTED** [review]

**Comment by scottj97** [comment]
@daniellustig would you agree that PTE.PBMT=3, being a reserved value, should cause a page fault?

**Comment by daniellustig** [comment]
> @daniellustig would you agree that PTE.PBMT=3, being a reserved value, should cause a page fault?

Yes, that matches the spec wording in the virtual address translation process: "...if any bits or encodings that are reserved for
future standard use are set within $pte$, stop and raise a page-fault exception corresponding to the original access type."

**Review by scottj97 — APPROVED** [review]

**Review by scottj97 — CHANGES_REQUESTED** [review]
Internal discussions revealed some architectural problems with this implementation. @rbuchner-aril is working on it.

**Review by scottj97 — APPROVED** [review]
As of ccfeaa99732479b029ce95fc9b09968c7f3b1f64 this looks in good shape to me.

**Review by aswaterman — APPROVED** [review]
LGTM.

**Pull request merged**